api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -123,6 +123,9 @@ message OAuth2Config {
   // Defines how ``client_id`` and ``client_secret`` are sent in OAuth client to OAuth server requests.
   // RFC https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
   AuthType auth_type = 11 [(validate.rules).enum = {defined_only: true}];
+
+  // Attempt to update access token using by refresh token if that is provided by authorization_endpoint.
+  bool use_refresh_token = 12;
```

Comment:
Probably update this description to something like: 

"Set to true to enable automatic access token refresh using the associated refresh token, provided that the OAuth server supports this feature."